{
  "mitigation": "Do not extract archives from untrusted sources with the Python tarfile module. Users of the module should add sanity checks when calling the tarfile.extract or tarfile.extractall functions.",
  "affected_release": [
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "release_date": "2023-11-14T00:00:00Z",
      "advisory": "RHSA-2023:6914",
      "package": "python3.11-pip-0:22.3.1-4.el8",
      "cpe": "cpe:/a:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "release_date": "2023-11-14T00:00:00Z",
      "advisory": "RHSA-2023:7024",
      "package": "python3.11-0:3.11.5-1.el8_9",
      "cpe": "cpe:/a:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "release_date": "2023-11-14T00:00:00Z",
      "advisory": "RHSA-2023:7034",
      "package": "python39:3.9-8090020230922213827.7484f1d1",
      "cpe": "cpe:/a:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "release_date": "2023-11-14T00:00:00Z",
      "advisory": "RHSA-2023:7034",
      "package": "python39-devel:3.9-8090020230922213827.7484f1d1",
      "cpe": "cpe:/a:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "release_date": "2023-11-14T00:00:00Z",
      "advisory": "RHSA-2023:7050",
      "package": "python38:3.8-8090020230810143931.d9f72c26",
      "cpe": "cpe:/a:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "release_date": "2023-11-14T00:00:00Z",
      "advisory": "RHSA-2023:7050",
      "package": "python38-devel:3.8-8090020230810143931.d9f72c26",
      "cpe": "cpe:/a:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "release_date": "2023-11-14T00:00:00Z",
      "advisory": "RHSA-2023:7151",
      "package": "python3-0:3.6.8-56.el8_9",
      "cpe": "cpe:/a:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "release_date": "2023-11-14T00:00:00Z",
      "advisory": "RHSA-2023:7176",
      "package": "python-pip-0:9.0.3-23.el8",
      "cpe": "cpe:/a:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "release_date": "2023-11-14T00:00:00Z",
      "advisory": "RHSA-2023:7151",
      "package": "python3-0:3.6.8-56.el8_9",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "release_date": "2023-11-14T00:00:00Z",
      "advisory": "RHSA-2023:7176",
      "package": "python-pip-0:9.0.3-23.el8",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8.6 Extended Update Support",
      "release_date": "2024-01-23T00:00:00Z",
      "advisory": "RHSA-2024:0374",
      "package": "python-pip-0:9.0.3-22.1.el8_6",
      "cpe": "cpe:/a:redhat:rhel_eus:8.6"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8.6 Extended Update Support",
      "release_date": "2024-01-25T00:00:00Z",
      "advisory": "RHSA-2024:0430",
      "package": "python3-0:3.6.8-47.el8_6.4",
      "cpe": "cpe:/a:redhat:rhel_eus:8.6"
    },
    {
      "product_name": "Red Hat Enterprise Linux 9",
      "release_date": "2023-11-07T00:00:00Z",
      "advisory": "RHSA-2023:6324",
      "package": "python3.11-pip-0:22.3.1-4.el9",
      "cpe": "cpe:/a:redhat:enterprise_linux:9"
    },
    {
      "product_name": "Red Hat Enterprise Linux 9",
      "release_date": "2023-11-07T00:00:00Z",
      "advisory": "RHSA-2023:6494",
      "package": "python3.11-0:3.11.5-1.el9_3",
      "cpe": "cpe:/a:redhat:enterprise_linux:9"
    },
    {
      "product_name": "Red Hat Enterprise Linux 9",
      "release_date": "2023-11-07T00:00:00Z",
      "advisory": "RHSA-2023:6659",
      "package": "python3.9-0:3.9.18-1.el9_3",
      "cpe": "cpe:/a:redhat:enterprise_linux:9"
    },
    {
      "product_name": "Red Hat Enterprise Linux 9",
      "release_date": "2023-11-07T00:00:00Z",
      "advisory": "RHSA-2023:6694",
      "package": "python-pip-0:21.2.3-7.el9",
      "cpe": "cpe:/a:redhat:enterprise_linux:9"
    },
    {
      "product_name": "Red Hat Enterprise Linux 9",
      "release_date": "2023-11-07T00:00:00Z",
      "advisory": "RHSA-2023:6659",
      "package": "python3.9-0:3.9.18-1.el9_3",
      "cpe": "cpe:/o:redhat:enterprise_linux:9"
    },
    {
      "product_name": "Red Hat Enterprise Linux 9",
      "release_date": "2023-11-07T00:00:00Z",
      "advisory": "RHSA-2023:6694",
      "package": "python-pip-0:21.2.3-7.el9",
      "cpe": "cpe:/o:redhat:enterprise_linux:9"
    },
    {
      "product_name": "Red Hat Software Collections for Red Hat Enterprise Linux 7",
      "release_date": "2023-11-08T00:00:00Z",
      "advisory": "RHSA-2023:6793",
      "package": "rh-python38-python-0:3.8.18-2.el7",
      "cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
    },
    {
      "product_name": "Red Hat Software Collections for Red Hat Enterprise Linux 7",
      "release_date": "2023-11-08T00:00:00Z",
      "advisory": "RHSA-2023:6793",
      "package": "rh-python38-python-pip-0:19.3.1-4.el7",
      "cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
    }
  ],
  "package_state": [
    {
      "product_name": "Red Hat Enterprise Linux 6",
      "fix_state": "Out of support scope",
      "package_name": "python",
      "cpe": "cpe:/o:redhat:enterprise_linux:6"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "fix_state": "Out of support scope",
      "package_name": "python",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "fix_state": "Out of support scope",
      "package_name": "python3",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Will not fix",
      "package_name": "python27:2.7/python2",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Not affected",
      "package_name": "python36:3.6/python36",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    }
  ],
  "threat_severity": "Moderate",
  "public_date": "2007-08-24T00:00:00Z",
  "bugzilla": {
    "description": "python: tarfile module directory traversal",
    "id": "263261",
    "url": "https://bugzilla.redhat.com/show_bug.cgi?id=263261"
  },
  "cvss": {
    "cvss_base_score": "",
    "cvss_scoring_vector": "",
    "status": ""
  },
  "cvss3": {
    "cvss3_base_score": "5.5",
    "cvss3_scoring_vector": "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:N/I:H/A:N",
    "status": "verified"
  },
  "iava": "",
  "cwe": "CWE-22",
  "statement": "The Red Hat Product Security has rated this issue as having a Moderate security impact, a future update may address this flaw. More information regarding issue severity can be found here: https://access.redhat.com/security/updates/classification.\nVersions of `python36:3.6/python36` as shipped with Red Hat Enterprise Linux 8 are marked as 'Not affected' as they just provide \"symlinks\" to the main `python3` component, which provides the actual interpreter of the Python programming language.",
  "acknowledgement": "",
  "name": "CVE-2007-4559",
  "document_distribution": "",
  "details": [
    "Directory traversal vulnerability in the (1) extract and (2) extractall functions in the tarfile module in Python allows user-assisted remote attackers to overwrite arbitrary files via a .. (dot dot) sequence in filenames in a TAR archive, a related issue to CVE-2001-1267.",
    "A flaw was found in the Python tarfile module. Extracting a crafted TAR archive with the tarfile.extract or tarfile.extractall functions could lead to a directory traversal vulnerability, resulting in overwrite of arbitrary files."
  ],
  "references": [
    "https://www.cve.org/CVERecord?id=CVE-2007-4559\nhttps://nvd.nist.gov/vuln/detail/CVE-2007-4559"
  ]
}